manifests/osfamily/redhat.pp

```diff
@@ -81,10 +81,12 @@
   }
 
   # Given the path to a key, see if it is imported, if not, import it
+  $legacy_gpg_pubkey = "gpg-pubkey-$(echo $(gpg --with-colons ${legacy_gpg_path} 2>&1 | grep ^pub | awk -F ':' '{print \$5}' | cut --characters=9-16 | tr [:upper:] [:lower:]))"
+
   exec {  "import-${legacy_keyname}":
     path      => '/bin:/usr/bin:/sbin:/usr/sbin',
     command   => "rpm --import ${legacy_gpg_path}",
-    unless    => "rpm -q gpg-pubkey-`echo $(gpg --throw-keyids < ${legacy_gpg_path}) | cut --characters=11-18 | tr '[:upper:]' '[:lower:]'`",
+    unless    => "rpm -q ${legacy_gpg_pubkey}",
```

Comment:
What does --throw-keyids do such that we no longer need it?

Reply:
--throw-keyids is actually an option that is used in the context of encrypting a message. In indicates whether they recipient key ids are included in the message metadata. Leaving them out (with --no-throw-keys) can provide better privacy when trying to communicate anonymously. My guess is it may have been an option with a side effect that worked in this situation, but it's no longer necessary.

That said, while doing some digging with git blame to answer this I noticed something that I should improve, so thanks for asking! :)